components/shared/DayRangePicker.tsx

```diff
@@ -0,0 +1,45 @@
+import { useRouter } from "next/router";
+import SingleSelect from "components/atoms/Select/single-select";
+
+export type DayRange = 7 | 30 | 90;
+
+const dayRanges: { label: `${DayRange}d`; value: `${DayRange}` }[] = [
+  {
+    label: "7d",
+    value: "7",
+  },
+  {
+    label: "30d",
+    value: "30",
+  },
+  {
+    label: "90d",
+    value: "90",
+  },
+];
+
+interface DayRangePickerProps {
+  onDayRangeChanged?: (value: string) => void;
+}
+
+export const DayRangePicker = ({ onDayRangeChanged }: DayRangePickerProps) => {
+  const router = useRouter();
+  const dayRange = (Number(router.query.range) || 30) as DayRange;
```

Comment:
I'm curious on your thoughts @brandonroberts @jpmcb about this design decision. In other places we pass in the range or other query string param value to a filter component, but in this case I'm not doing that.

The reasoning being is the job of this component is to allow you to pick a day range and then it adds the day range to the querystring of the page it's currently on. This way, it's the pages responsibility to handle the updated query string value, `range` in this case to load data based on that value.

TLDR; it forces URL driven development. 😎

Reply:
I think its reasonable as it only controls a query param